### PR TITLE
Improve Graveler Delete by checking staging first and return when possible

### LINKED ISSUE #4193: Improve Graveler delete prevent access to committed when possible
Improve Graveler Delete operation by returning as soon possible.
Checking staging first and returning if needed will perform faster than checking both committed and uncommitted all the time.

### PR DESCRIPTION
**Pull request by nopcoder**
Improve Graveler Delete operation by returning as soon as possible.

Close #4193

### CODE REVIEW

File: pkg/graveler/graveler.go [hunk 1]
```diff
@@ -1451,47 +1451,43 @@ func (g *KVGraveler) Delete(ctx context.Context, repository *RepositoryRecord, b
...
-			foundInCommitted = true
-		}
 
-		foundInStaging := true
-		val, err := g.getFromStagingArea(ctx, branch, key)
-		if err != nil {
-			if !errors.Is(err, ErrNotFound) {
-				// unknown error
-				return fmt.Errorf("reading from staging: %w", err)
+			// check staging for entry or tombstone
```
Comment (itaiad200) [inline pkg/graveler/graveler.go]: Can be performed after the staging read.

File: pkg/graveler/graveler.go [hunk 2]
```diff
@@ -1451,47 +1451,42 @@ func (g *KVGraveler) Delete(ctx context.Context, repository *RepositoryRecord, b
...
+			case err != nil:
+				if !errors.Is(err, ErrNotFound) {
+					// unknown error
+					return fmt.Errorf("reading from staging: %w", err)
+				}
+				// entry not found in staging - continue to committed check
+			case val == nil:
+				// found tombstone in staging, return ErrNotFound
+				return ErrNotFound
+			default:
```
Comment (itaiad200) [inline pkg/graveler/graveler.go:1468]: Consider doing it explicit for readability:
```suggestion
			case val != nil:
```

**Review by itaiad200 — APPROVED** [review]

File: pkg/graveler/graveler.go (hunk 1 above)
Comment (nopcoder) [reply pkg/graveler/graveler.go]: Done - thanks

File: pkg/graveler/graveler.go (hunk 2 above)
Comment (nopcoder) [reply pkg/graveler/graveler.go:1468]: prefer to catch all at this point

**Pull request merged**